app/src/main/java/cm/aptoide/pt/ads/MoPubAnalytics.java

```diff
@@ -12,4 +12,39 @@ public void setMoPubAbTestGroup(boolean isControlGroup) {
     AppEventsLogger.updateUserProperties(bundle, response -> Logger.getInstance()
         .d("Facebook Analytics: ", response.toString()));
   }
+
+  public void setAdsVisibilityUserProperty(
+      WalletAdsOfferManager.OfferResponseStatus offerResponseStatus) {
+    Bundle bundle = new Bundle();
+    bundle.putString("ads", mapToAdsVisibility(offerResponseStatus).getType());
+    AppEventsLogger.updateUserProperties(bundle, response -> Logger.getInstance()
+        .d("Facebook Analytics: ", response.toString()));
+  }
+
+  private AdsVisibility mapToAdsVisibility(WalletAdsOfferManager.OfferResponseStatus status) {
+    switch (status) {
+      case NO_ADS:
+        return AdsVisibility.CONTROL;
+      case ADS_HIDE:
+        return AdsVisibility.ADS_BLOCKED_BY_OFFER;
+      case ADS_SHOW:
+        return AdsVisibility.HAS_ADS;
+      default:
+        return AdsVisibility.CONTROL;
```

Comment:
If this shouldn't happen I would throw an IllegalStateException or an error so we don't pollute analytics with data that could be not correct